src/nccl_ofi_net.c

```diff
@@ -1543,6 +1580,28 @@ static ncclResult_t ofi_accept(void *listenComm, void **recvComm)
 			goto exit;
 	}
 
+	/* Get local EP address to detect requests to self */
+	ret = fi_getname(&(nccl_ofi_component[dev]->ep->fid),
+	                 (void *)&local_ep_addr,
+	                 &namelen);
+	if (ret != 0) {
+		NCCL_OFI_WARN("Call to fi_getname() failed with RC: %d, ERROR: %s",
+		              ret, fi_strerror(-ret));
+		ret = ncclSystemError;
+		goto error;
+	}
+
+	if (memcmp(local_ep_addr, remote_ep_addr, namelen) == 0) {
+		NCCL_OFI_TRACE(NCCL_NET, "Accept from self");
+		if (request_to_self != NULL) {
```

Comment:
Shouldn't the code assert for this? If we get a remote address then `request_to_self` should have a valid value right?

Reply:
If by some reason due to network conditions block of data would be duplicated -- we prevent null pointer dereference by free_nccl_ofi_req(). As network conditions are inherently runtime-only it would be a little bit optimistic to rely upon assert here.

Reply:
Umm...shouldn't the provider itself handle the duplicate rather than sending it to application? As a user of libfabric, we expect only one completion and one recv for every message posted. It is apps responsibility to post another receive buffer if they would like to get some other data.

Reply:
Thanks for suggestion, the fix has been reworked from scratch.